### PR TITLE
Add NOOP action reward during training

### PR DESCRIPTION
**Pull request by taufeeque9**

### CODE REVIEW

File: cleanba/load_and_eval.py [hunk 1]
```diff
@@ -94,17 +97,21 @@ def recursive_find_checkpoint(root: Path) -> Iterable[Path]:
...
-    checkpoints_to_load: List[Tuple[int, Path]] = []
-    for cp_candidate in recursive_find_checkpoint(args.load_other_run):
-        match = cp_expr.match(str(cp_candidate))
-        if match is None:
-            print("Skipping (not matching)", cp_candidate)
-        else:
-            checkpoints_to_load.append((int(match.group(1)), cp_candidate))
+    if args.checkpoints_to_load:
+        assert args.only_last_checkpoint is False, "Can't specify both checkpoints_to_load and only_last_checkpoint."
+        checkpoints_to_load = [(int(cp_expr.match(cp).group(1)), args.load_other_run / cp) for cp in args.checkpoints_to_load]  # type: ignore
```
Comment (rhaps0dy) [inline cleanba/load_and_eval.py:106]: I think there shouldn't be a `type: ignore` here, and you should specify the type outside of the if/else. That would fix it.

**Review by rhaps0dy — CHANGES_REQUESTED** [review]
Pretty great!, but I think some things need to change.

File: cleanba/environments.py [hunk 2]
```diff
@@ -194,6 +202,9 @@ def __init__(self, env: gym.vector.VectorEnv):
         self.observation_space = batch_space(self.single_observation_space, n=self.num_envs)
         self.single_action_space = env.single_action_space
         self.action_space = env.action_space
+        if remove_last_action:
```
Comment (rhaps0dy) [inline cleanba/environments.py:205]: rather than modifying, I'd trust constructing a new Discrete action space more. What if there's state other than the `.n`? It's not the case but the code reader cannot easily tell.

```suggestion
        if remove_last_action:
            assert isinstance(self.single_action_space, gym.spaces.Discrete)
            self.single_action_space = gym.spaces.Discrete(self.single_action_space - 1)
            self.action_space = batch_space(self.single_action_space, n=self.num_envs) 
```

File: cleanba/evaluate.py [hunk 3]
```diff
@@ -51,10 +58,14 @@ def run(self, policy: Policy, get_action_fn, params, *, key: jnp.ndarray) -> dic
                     episode_success = np.zeros(envs.num_envs, dtype=np.bool_)
                     episode_returns = np.zeros(envs.num_envs, dtype=np.float64)
                     episode_lengths = np.zeros(envs.num_envs, dtype=np.int64)
+                    episode_acts = np.zeros((max_steps, envs.num_envs), dtype=np.int32)
+                    episode_rewards = np.zeros((max_steps, envs.num_envs), dtype=np.float32)
 
+                    last_box_time_step = -1 * np.ones(envs.num_envs, dtype=np.int64)
+                    noops_array = np.zeros((envs.num_envs, max_steps), dtype=bool)
                     i = 0
+                    all_obs = [obs]
```
[the remark this answers is not in the document]
Comment (rhaps0dy) [reply cleanba/evaluate.py:68]: Maybe name it `this_minibatch_obs`.

**Review by rhaps0dy — APPROVED** [review]
LGTM though note comments.
